### PR TITLE
refactor(xml-helpers): DRY & simplify

### PR DESCRIPTION
**Pull request by raphinesse**
### Motivation and Context

Refactor triggered by me reviewing #112

### Description

Simplification without functional changes

### Testing

`npm t`

### CODE REVIEW

File: src/util/xml-helpers.js [hunk 1]
```diff
@@ -163,82 +115,73 @@ module.exports = {
...
-        } else {
-            parent = doc.find(selector);
-        }
-        return parent;
+        if (!selector.startsWith('/')) return doc.find(selector);
+
+        // elementtree does not implement absolute selectors so we build an
+        // extended tree where we can use an equivalent relative selector
+        const überRoot = et.Element('über-root');
+        überRoot.append(doc.getroot());
```
Comment (erisu) [inline src/util/xml-helpers.js]: `ü` a wrong `u`?

File: src/util/xml-helpers.js [hunk 2]
```diff
@@ -163,82 +115,73 @@ module.exports = {
...
+        return überRoot.find(`.${selector}`);
```
Comment (erisu) [inline src/util/xml-helpers.js]: Same here

File: src/util/xml-helpers.js (hunk 1 above)
Comment (raphinesse) [reply src/util/xml-helpers.js]: No, it's the [right `u`](https://en.wiktionary.org/wiki/uber) 😋. This was just meant as an Easter egg for the review.

We should change it to `metaRoot` for production.

Comment (erisu) [reply src/util/xml-helpers.js]:
```suggestion
        metaRoot.append(doc.getroot());
```

File: src/util/xml-helpers.js [hunk 3]
```diff
@@ -163,82 +115,73 @@ module.exports = {
...
-            }
-        } else {
-            parent = doc.find(selector);
-        }
-        return parent;
+        if (!selector.startsWith('/')) return doc.find(selector);
+
+        // elementtree does not implement absolute selectors so we build an
+        // extended tree where we can use an equivalent relative selector
+        const überRoot = et.Element('über-root');
```
Comment (erisu) [inline src/util/xml-helpers.js]:
```suggestion
        const metaRoot = et.Element('meta-root');
```

File: src/util/xml-helpers.js (hunk 2 above)
Comment (erisu) [inline src/util/xml-helpers.js]:
```suggestion
        return metaRoot.find(`.${selector}`);
```

**Pull request merged**